File: original_examples/Codes4Unit5/norcau.m
%norcau
%
x=2;
N=100000;
num = 0; denom = 0;
%
for i=1:N
theta = x + randn;  %theta ~ N(x,1)
num = num + theta/(1+theta^2);
denom = denom + 1/(1+theta^2);
end
delta21 = num/denom

%%

x=2;
N=100000;
num = 0; denom = 0;
%
for i=1:N
a=randn(2,1);
theta = a(1)/a(2);  %theta ~ Ca(0,1)
num = num + theta * exp(-1/2 *(x-theta)^2);
denom = denom + exp(-1/2 *(x-theta)^2);
end
delta22 = num/denom

File: original_examples/Codes4Unit5/norcau.py
g# -*- coding: utf-8 -*-
"""
Bayes estimator delta(x) for x=2,
for Normal-Cauchy Model
Created on Thu Dec 28 11:56:56 2017
@author: bv20
"""

from scipy import integrate, inf, exp
x = 2
num = lambda th: th * exp(-0.5*(x-th)**2)/(1+th**2)
denom = lambda th: exp(-0.5*(x-th)**2)/(1+th**2) 
delta2 = integrate.quad(num,-inf,inf)[0]/integrate.quad(denom,-inf,inf)[0]
#delta(2)
print(delta2) #1.2821951026935339

# Errors

numerator =integrate.quad(num,-inf,inf)[0]    #0.9159546679977636
denominator=integrate.quad(denom,-inf,inf)[0] #0.714364503556127
errnum=integrate.quad(num,-inf,inf)[1]        #1.0415234856193602e-09
errdenom=integrate.quad(denom,-inf,inf)[1]    #1.2022419107752649e-08

err = delta2 * (errnum/numerator + errdenom/denominator)
print(err)   #2.3036713479165735e-08